### ISSUE TITLE
Wire up favicons in the reworked Bookmarks panel

### ISSUE THREAD

**Issue by justindarc** [mozilla-mobile/firefox-ios#4776]
Once #4739 lands on `rust-dev`, we'll need to wire up favicons in the newly-reworked Bookmarks panel. They are not going to be included as part of #4739 for the sake of breaking out the work into its own PR.
